### ISSUE TITLE
[scaffolder-backend] Export SecureTemplater to enable CI/CD based templating for testing

### ISSUE THREAD

**Issue by ScheerMT** [backstage/backstage#9343]
## Feature Suggestion
To accomplish CI/CD of a template project outside of the Backstage ecosystem (for maintainers of these templates), a method to generate the final templates in place needs to be made available. This functionality should be run directly against the template project and optionally in place replace the project files with the final templated state. 

I am not necessarily suggesting building out an extremely robust solution here to start! I would like the necessary foundational pieces of making the generation of a template with a values file be made available outside of the Backstage Software Template feature even if it means rolling the driver program myself.

## Possible Implementation
I have two ideas that are incremental on each other if I understand the classes sufficiently well and gets friendlier for end users.

### Export SecureTemplater
I took some time reading through the `scaffolder-backend` and while I am not proficient in the javascript environment, the pieces seem to be available in some form within the [SecureTemplater](https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts#L116) class which is not current exported but could be at the cost of additional API surface area and support overhead.  The functionality here seems like you are able to pass in an arbitrary string needing to be templated along with the values to replace and it will output the templated string which gets me unblocked from moving forward with rolling my driver program (I think?)

Bonus points for this implementation if the `SecureTemplater` and its direct dependencies were to be broken out into its own NPM package to shave down overall module size for use cases like my own.

### Extend NunjucksWorkflowRunner
With my thimble-sized understanding of workflows/tasks within this plugin, An alternative and one layer higher would be to enable this Runner to point to a set of local files needing templating and let that be the entrypoint for the CI script to be run. This sounds like it _could_ be the better option long term but ends up being more upfront work too.

## Context
While maintaining templates utilized for Backstage's Software Templates, an important aspect is ensuring the template project itself remains correct via PR tests that exercise different aspects (like compiling or deploying) of the project. In order to do this, while remaining aligned with Backstage's assumptions on how templating _actually_ works, a method to generate the templated project outside of Backstage needs to be available.

**Comment by Crevil** [comment]
It indeed seem to hit the use case I had in mind with this 👍

**Comment by Rugvip** [comment]
@ScheerMT Gonna close as It seems #6568 is a duplicate of the end goal of this, and there's more discussion there. Lets us know if you'd like to re-open
